[stdlib] Document that removeAll(where:) doesn't reorder the remaining elements.

krilnon opened the pull request:
Per Swift Forums discussion on the topic:
[Does removeAll(where:) on arrays guarantee preserved order of elements?](https://forums.swift.org/t/does-removeall-where-on-arrays-guarantee-preserved-order-of-elements/15313)

Yes it does; document it.

krilnon committed:
Document that removeAll(where:) doesn't reorder the remaining elements.

Per Swift Forums discussion on the topic:
"Does removeAll(where:) on arrays guarantee preserved order of elements?"

krilnon commented:
@swift-ci Please smoke test

natecook1000 commented:
Would you mind copying the docs from the first default implementation up to the protocol requirement? Otherwise LGTM!

krilnon committed:
Copy docs from the first default impl up to the protocol requirement.

Per Nate's feedback on PR#18803.

The protocol requirement was missing a few extra sentences and a
code example.

krilnon committed:
Fix a code example that would fail to remove negative odd integers.

Per feedback on the forums from Jens,
swap `== 1` with `!= 0` so that negative odd numbers would be
removed, too, if the example were modified to include some negative
numbers in the input array.

krilnon commented:
@swift-ci Please smoke test

krilnon commented:
@natecook1000 Okay; did that. I also swapped the oddness checking with what Jens suggested in the forum thread.
